### PR TITLE
Bug 1865357 - Add Global Privacy Control settings

### PR DESCRIPTION
**Pull request by mcarare**
https://www.figma.com/file/RyIADLmC3B0BIS9jSiAti3/Settings?node-id=0%3A2212&mode=dev#629804471
The option is available regardless of the selected ETP option

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1865357

### CODE REVIEW

**Comment by mcarare** [comment]
Adding @AndiAJ & @sv-ohorvath for UI tests review.

File: android-components/components/concept/engine/src/main/java/mozilla/components/concept/engine/Settings.kt [hunk 1]
```diff
@@ -241,6 +241,11 @@ abstract class Settings {
      * Setting the HTTPS-Only mode for upgrading connections to HTTPS.
      */
     open var httpsOnlyMode: Engine.HttpsOnlyMode by UnsupportedSetting()
+
+    /**
+     * Setting o control whether Global Privacy Control isenabled.
```
Comment (Amejia481) [inline android-components/components/concept/engine/src/main/java/mozilla/components/concept/engine/Settings.kt:246]: nit: Setting **to** control whether Global Privacy Control **is** enabled.

**Review by Amejia481 — APPROVED** [review]
LGTM!
Only one documentation nit!

File: fenix/app/src/androidTest/java/org/mozilla/fenix/ui/robots/SettingsSubMenuEnhancedTrackingProtectionRobot.kt [hunk 2]
```diff
@@ -66,6 +69,36 @@ class SettingsSubMenuEnhancedTrackingProtectionRobot {
...
 
+    fun scrollToGCPSettings(): ViewInteraction = onView(withId(R.id.recycler_view)).perform(
+        RecyclerViewActions.scrollTo<RecyclerView.ViewHolder>(
+            hasDescendant(withText(globalPrivacyControlSwitchText)),
+        ),
+    )
+    fun verifyGPCTextWithSwitchWidget() {
+        onView(
+            allOf(
+                withParentIndex(1),
```
Comment (sv-ohorvath) [inline fenix/app/src/androidTest/java/org/mozilla/fenix/ui/robots/SettingsSubMenuEnhancedTrackingProtectionRobot.kt:80]: Only one small change: we usually don't use indexes as they can change easily. I think in this case, just the text selectors should be enough.

File: android-components/components/concept/engine/src/main/java/mozilla/components/concept/engine/Settings.kt (hunk 1 above)
Comment (mcarare) [reply android-components/components/concept/engine/src/main/java/mozilla/components/concept/engine/Settings.kt]: 🦅 👁️

**Review by AndiAJ — APPROVED** [review]
👍🏽

**Pull request merged**